contracts/VeJoeStaking.sol

```diff
@@ -104,12 +154,23 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
         emit UpdateMaxCap(msg.sender, _maxCap);
     }
 
-    /// @notice Set veJoePerSec
-    /// @param _veJoePerSec The new veJoePerSec
-    function setVeJoePerSec(uint256 _veJoePerSec) external onlyOwner {
+    /// @notice Set veJoePerSharePerSec
+    /// @param _veJoePerSharePerSec The new veJoePerSharePerSec
+    function setVeJoePerSharePerSec(uint256 _veJoePerSharePerSec) external onlyOwner {
         updateRewardVars();
-        veJoePerSec = _veJoePerSec;
-        emit UpdateVeJoePerSec(msg.sender, _veJoePerSec);
+        veJoePerSharePerSec = _veJoePerSharePerSec;
```

Comment:
Probably good to have some upper bound on this (it can be high), it won't permanently break the contract if we set it stupidly high but could cause overflow on line 265, 
`secondsElapsed.mul(veJoePerSharePerSec).mul(ACC_VEJOE_PER_SHARE_PRECISION)` could cause phantom overflow.

Reply:
See phantom overflow in this great series of articles [math in solidity](https://medium.com/coinmonks/math-in-solidity-part-2-overflow-3cd7283714b4)

Reply:
Added a `upperLimitVeJoePerSharePerSec = 1e36`. thanks for the article, was a good read!